src/hooks/useCalculateIncentives.tsx

```diff
@@ -193,15 +193,18 @@ export const useCalculateCurrentAccountCurrentBsxReceived = () => {
     const { data: { contributions } } = useAccount();
     const { data: { mostRecentAuctionClosingStart } } = useChronicle();
     const { data: { totalContributionWeight } } = useIncentives();
+    const { parachain: { data: { fundsPledged: contributionWeightOfAllContributors } } } = useOwn();
+
+    console.log(totalContributionWeight)
+    console.log(contributionWeightOfAllContributors)
 
     return calculateCurrentBsxReceived(
         contributions, 
         mostRecentAuctionClosingStart,
-        totalContributionWeight
+        `${contributionWeightOfAllContributors}000000`
```

Comment:
`totalContributioWeight` (220899542720391358000000) from `useIncentives` is wrong.  I don't know why.

`contributionWeightOfAllContributors` (222221915331441358) from `useOwn` is correct.

☝️ this produces the correct amount of BSX rewards with every account I checked.
- All contributions get a multiplier of 1.  They all contributed before the ending-period
  - can see last contribution here: https://kusama.subscan.io/crowdloan/2090-35?tab=contributor 
  - can see ending-period-start block here: https://kusama.subscan.io/auction/8
    - according to those links,  the last contribution happened at `9 251 619` and the ending period started at `9 262 576`
- I manually checked this fix against the rewards formula
  - using `15000000000 / 222221.915331441358 * contributorWeight` for three different accounts
  - the output of this fix and the formula match the three accounts I checked
- I also compared this fix with the rewards.json script results: https://raw.githubusercontent.com/galacticcouncil/Basilisk-crowdloan-ui/main/scripts/data/rewards.json
  - the results in the UI with this fix,  and the results from the script ☝️,  match

All this can mean that we don't need to hard code in a multiplier of `1` after all.  Thoughts?

Reply:
@jak-pan ^ all of our contributors are getting a multiplier of 1, due to how the auctions went.

Reply:
@timothydm42 are you saying that the UI calculates the BSX multiplier correctly, bust just the totalContributionWeight was wrong? If thats the case, then the total contribution weight should equal the amount raised since all contributions get a weight of 1. Please investigate when we have stopped the processor (via chronicle.lastBlockProcessed) and check which event triggers the reset of the total contribution weight. I am 79.5% certain we've stopped it before the multiplier reset on the backend occured and thats why the frontend gets wrong data. 

Which still means you will just end up replacing the total contribution weight with a mock weight of raised * 1, but it'd be good to be certain on what happened on the processor side.

Reply:
@timothydm42 @maht0rz 

just one comment. The total weight is correct but the problem is here:

https://github.com/galacticcouncil/Basilisk-crowdloan-ui/blob/e1e2e4787312811d52aba2dda7d1533d61e9c237/src/hooks/useCalculateIncentives.tsx#L123

Within the `calculateCurrentBsxReceived `- the total is divided by precision again for some reason - when I saw first time - i was wondering but I haven't investigated why it is needed to do so.

That's the reason why the result is correct when you just append 000000 becuse the first operation is division by 1_000_000 ,so you get back the same total weight.

Reply:
Ignore that deleted comment by me. I misunderstood.

@maht0rz  regarding the difference in total weights -  isn't the reason that the multipliers for last entries were not 1 but slightly less ? You remember that i noticed that while processing the data.

Reply:
```
Please investigate when we have stopped the processor (via chronicle.lastBlockProcessed) and check which event triggers the reset of the total contribution weight. I am 79.5% certain we've stopped it before the multiplier reset on the backend occured and thats why the frontend gets wrong data.
```
@maht0rz  re: 👆 

I don't see anything wrong with the `blockheight` at which we stopped the processor,  because we stopped it _after_ the previous parachain slot was leased:

- We [reset](https://github.com/maht0rz/basilisk-crowdloan-query/blob/develop/mappings/slots/handleSlotsLeased.ts#L6) the _multiplier_ after the `[slot(Leased)](https://github.com/maht0rz/basilisk-crowdloan-query/blob/develop/manifest.yml#L66)` event.  The last `slot(Leased)` event _before_ our win is contained in [9 233 727](https://kusama.subscan.io/event?module=slots&event=leased)

- we stopped the processor after that,  at `9 263 121`
  - ![image](https://user-images.githubusercontent.com/25761350/142251194-bc36ca8f-c609-48ab-bef9-22eb0b86a4c4.png)

nothing in the reset logic looks off to me... and @enthusiastmartin I haven't found a way to get historical `totalContributionWeight`s through the processor so that I can look at them over time.

what do you guys think?